lib/portage/package/ebuild/fetch.py

```diff
@@ -201,6 +226,28 @@ def _userpriv_test_write_file(settings, file_path):
     if rval is not None:
         return rval
 
+    # Return early in order to avoid an unecessary spawn and event loop
+    # recursion when privileges do not need to be dropped.
+    if not (
+        "userfetch" in settings.features
+        and os.getuid() == 0
+        and portage_gid
+        and portage_uid
+        and hasattr(os, "setgroups")
+    ):
+        rval = os.access(os.path.dirname(file_path), os.W_OK)
+        _userpriv_test_write_file_cache[file_path] = rval
+        return rval
+
+    if asyncio.get_event_loop().is_running():
+        # Approximate the permission test in order to avoid this error:
+        # RuntimeError: This event loop is already running
+        # FIXME: This approximation only supports Unix permissions and ignores ACLs.
+        st = os.stat(os.path.dirname(file_path))
+        rval = st.st_gid == int(portage_gid) and stat.S_IMODE(st.st_mode) & 0o020
+        _userpriv_test_write_file_cache[file_path] = rval
+        return rval
+
```

Comment:
Since this approximation doesn't support ACLs, we could possibly use it just as an optimization for the case where the `st.st_gid == int(portage_gid) and stat.S_IMODE(st.st_mode) & 0o020` expression holds (ACLs no used), and if the result of this expression is `False` then proceed with the spawn in order to account for ACLs.

Reply:
The function is now async, so I've removed the `loop.is_running()` check and converted this Unix permission code to a conditional optimization:
```python
    # Optimize away the spawn if we can detect a portage group write
    # permission bit, but if this optimization fails then continue with
    # the spawn for ACL support.
    st = os.stat(os.path.dirname(file_path))
    if st.st_gid == int(portage_gid) and stat.S_IMODE(st.st_mode) & 0o020:
        _userpriv_test_write_file_cache[file_path] = True
        return True
```

Reply:
It turns out that the approximation / optimization to check for the portage group write bit is not needed, since we can simply use an `os.access` check in any case that the `_want_userfetch` function returns False. So, we have just this:
```python
    # Optimize away the spawn when privileges do not need to be dropped.
    if not _want_userfetch(settings):
        rval = os.access(os.path.dirname(file_path), os.W_OK)
        _userpriv_test_write_file_cache[file_path] = rval
        return rval
```